core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1104,6 +1104,16 @@ private SqlNode validateScopedExpression(
       List<SqlNode> operands = ((SqlCall) node).getOperandList();
       SqlSampleSpec sampleSpec = SqlLiteral.sampleValue(operands.get(1));
       if (sampleSpec instanceof SqlSampleSpec.SqlTableSampleSpec) {
+        // The sampling percentage is between 0(0%) and 1(100%),1 is exclusive
+        float samplePercentage =
```

Comment:
Not your fault, but it seems foolish to convert samplePercentag to a `float` and then back to a `BigDecimal`.

Let's keep it as a `BigDecimal`. Can you change the type of field `SqlTableSampleSpec.samplePercentage` from `float` to `BigDecimal`, and make it public. The method `getSamplePercentage()` should remain `float` but is deprecated.

Reply:
Hi @julianhyde ,I had the same idea with you to change samplePercentag to Bigdecimal. I thought that there was some reason to set samplePercentag to float before, so I did not change this code.

I will follow your advice today.